### DISCUSSION TITLE
Searching for definitions possibility

### DISCUSSION THREAD

**Discussion by susliko** [scalameta/nvim-metals#244] [Q&A] [answered]
Hi! I wonder if there are any methods in Metals, which would allow implementing a full-text search on workspace definitions (traits/classes/objects), like Intellij IDEA does.

Passing these to Telescope would be very sweet

![image](https://user-images.githubusercontent.com/15569000/136375004-4847fcfb-893d-4620-947c-4c08dabf933b.png)

**Comment by ckipp01** [comment, marked as answer]
Hey @susliko! Absolutely, what you're looking for is the [Workspace Symbol](https://microsoft.github.io/language-server-protocol/specification#workspace_symbol) search.

![2021-10-07 13 35 02](https://user-images.githubusercontent.com/13974112/136376715-bc7ec774-3caa-4a84-b65f-b82109a7a97a.gif)

By default this is the `vim.lsp.buf.workspace_symbol()`. If you want to use it in Telescope there is a bit more work since by default Telescope sends in an empty string to the server, but Metals won't respond to an empty string. Instead you can do something like this which I use in my config:

```lua
M.lsp_workspace_symbols = function()
  local input = vim.fn.input("Query: ")
  vim.api.nvim_command("normal :esc<CR>")
  if not input or #input == 0 then
    return
  end
  require("telescope.builtin").lsp_workspace_symbols({ query = input })
end
```
And then I use this function in a mapping
```lua
map("n", "gws", [[<cmd>lua require"settings.telescope".lsp_workspace_symbols()<CR>]])
```

**Reply by susliko** [reply]
Whoa! Thank you so much
